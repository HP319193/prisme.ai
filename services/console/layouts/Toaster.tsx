import {
  createContext,
  FC,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import { Toast } from "primereact/toast";

export const context = createContext<Toast>({} as Toast);

export const useToaster = () => useContext(context);

export const Toaster: FC = ({ children }) => {
  const ref = useRef(null);
  const [toaster, setToaster] = useState<any>();
  useEffect(() => {
    // Force toaster to re-render with ref
    setToaster(ref.current);
  }, [ref]);
  return (
    <context.Provider value={toaster!}>
      {children}
      <Toast ref={ref} position="bottom-right" />
    </context.Provider>
  );
};

export default Toaster;
