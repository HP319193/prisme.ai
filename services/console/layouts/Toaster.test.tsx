import Toaster from './Toaster';
import renderer from 'react-test-renderer';

it('should render', () => {
  const root = renderer.create(<Toaster />);
  expect(root.toJSON()).toMatchSnapshot();
});
