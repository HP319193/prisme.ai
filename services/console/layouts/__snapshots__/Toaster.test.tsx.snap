// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="p-toast p-component p-toast-bottom-right"
  id={null}
  style={null}
>
  <div />
</div>
`;
